Cython/Compiler/Main.py

```diff
@@ -60,26 +60,21 @@ class Context:
     #  future_directives     [object]
     #  language_level        int     currently 2 or 3 for Python 2/3
 
-    cython_scope = None
     language_level = None  # warn when not set but default to Py2
 
     def __init__(self, include_directories, compiler_directives, cpp=False,
                  language_level=None, options=None):
```

Comment:
After reading through most of the implementation again, I start to wonder if the `shared_utility_qualified_module_name` should be part of the `Context`. It's really a piece of information that is needed in order run the translation process, related to import paths etc. And it doesn't feel much like a module directive in that sense, it's an option that stands above the translation of a single module.

Reply:
I was thinking that it might avoid the signature changes where you pass in the module instead of the context. That reduces the dependencies and seems an advantage to me.